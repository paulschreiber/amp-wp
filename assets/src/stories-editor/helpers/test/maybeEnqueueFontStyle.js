/**
 * Internal dependencies
 */
import { maybeEnqueueFontStyle } from '../';

describe( 'maybeEnqueueFontStyle', () => {
	it( 'should ignore invalid font name', () => {
		expect( maybeEnqueueFontStyle( undefined ) ).toEqual( undefined );
	} );

	it( 'should ignore missing font name', () => {
		expect( maybeEnqueueFontStyle( 'Tahoma' ) ).toEqual( undefined );
	} );

	it( 'should ignore font without handle', () => {
		expect( maybeEnqueueFontStyle( 'Ubuntu' ) ).toEqual( undefined );
	} );

	it( 'should ignore font without src', () => {
		expect( maybeEnqueueFontStyle( 'Verdana' ) ).toEqual( undefined );
	} );

	it( 'should enqueue requested font only once', () => {
		maybeEnqueueFontStyle( 'Roboto' );
		maybeEnqueueFontStyle( 'Roboto' );

		expect( document.head ).toMatchSnapshot();
	} );
} );
