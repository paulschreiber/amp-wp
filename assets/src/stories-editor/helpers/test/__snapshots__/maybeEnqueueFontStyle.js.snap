// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maybeEnqueueFontStyle should enqueue requested font only once 1`] = `
<head>
  <style />
  <link
    crossorigin="anonymous"
    href="https://fonts.googleapis.com/css?family=Roboto%3A400%2C700&subset=latin%2Clatin-ext"
    id="roboto-font"
    media="all"
    rel="stylesheet"
    type="text/css"
  />
</head>
`;
